src/analyses/base.ml

```diff
@@ -2241,9 +2241,42 @@ struct
         (* List.map (fun f -> f (fun lv -> (fun x -> set ~ctx:(Some ctx) ctx.ask ctx.global st (eval_lv ctx.ask ctx.global st lv) (Cilfacade.typeOfLval lv) x))) (LF.effects_for f.vname args) |> BatList.fold_left D.meet st *)
       end
 
-  let combine ctx (lval: lval option) fexp (f: fundec) (args: exp list) fc (after: D.t) : D.t =
+  let combine_st ctx (local_st : store) (fun_st : store) (tainted_lvs : Q.LS.t) : store =
+    Q.LS.fold (fun (v, o) st -> 
+      if CPA.mem v fun_st.cpa then
+        let lval = Lval.CilLval.to_lval (v,o) in
+        let address = eval_lv (Analyses.ask_of_ctx ctx) ctx.global st lval in
+        let lval_type = (AD.get_type address) in
+        if M.tracing then M.trace "taintPC" "updating %a; type: %a\n" Lval.CilLval.pretty (v, o) d_type lval_type;
+        match CPA.find_opt v (fun_st.cpa) with
+        | None -> st
+        | Some (`Array _) when (get_string "ana.base.arrays.domain") = "partitioned" -> begin
+          (* partitioned arrays cannot be copied by individual lvalues, so if tainted just copy the whole callee value for the array variable *)
+          let new_arry_opt = CPA.find_opt v fun_st.cpa in
+          match new_arry_opt with
+          | None -> st
+          | Some new_arry -> {st with cpa = CPA.add v new_arry st.cpa}
+          end
```

Comment:
It would be possible to taint the whole array when a lval inside a partitioned array is written.
However special casing this inside the `taintPartialContext` analysis does not seem like a better approach, as there might be other analyses that can benefit from the finer taint information for partitioned arrays.

Other than that I was not able to come up with a better idea to address this issue.